### PR TITLE
Add Dynamic Sitemap

### PR DESCRIPTION
**Pull request by BarisSari**
- [ ] Generate a sitemap for all the HTML views

### CODE REVIEW

File: budgetportal/sitemaps.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            args=[
+                item.government.sphere.financial_year.slug,
+                item.government.sphere.slug,
+                item.government.slug,
+                item.slug,
+            ],
+        )
+
+
+class DepartmentPreviewJsonSitemap(sitemaps.Sitemap):
```
Comment (jbothma) [inline budgetportal/sitemaps.py]: Please remove this class.
Only want sitemap for HTML endpoints, not .json.

File: budgetportal/sitemaps.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+
+class InfrastructureProjectPartViewSitemap(sitemaps.Sitemap):
+    def items(self):
+        return InfrastructureProjectPart.objects.all()
+
+
+class FocusViewSitemap(sitemaps.Sitemap):
+    def items(self):
+        focus_slugs = []
```
Comment (jbothma) [inline budgetportal/sitemaps.py]:
```suggestion
        focus_area_page_params = []
```

Comment (jbothma) [reply budgetportal/sitemaps.py]: These aren't slugs, these are dicts.

File: budgetportal/sitemaps.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        for year in FinancialYear.get_available_years():
+            treemap = get_consolidated_expenditure_treemap(year)
+            for data in treemap["data"]["items"]:
+                focus_slugs.append({"year": str(year.slug), "focus": data["id"]})
```
Comment (jbothma) [inline budgetportal/sitemaps.py]: focus_area_page_params

File: budgetportal/sitemaps.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        return focus_slugs
+
+    def location(self, item):
+        return reverse("focus", args=[item["year"], item["focus"]])
```
Comment (jbothma) [inline budgetportal/sitemaps.py]: focus_area_page_params

File: budgetportal/sitemaps.py (hunk 2 above)
Comment (jbothma) [reply budgetportal/sitemaps.py]: Also fix the other referneces to this.

File: budgetportal/sitemaps.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+
+class FocusJsonViewSitemap(sitemaps.Sitemap):
```
Comment (jbothma) [inline budgetportal/sitemaps.py]: Please remove - don't want json views in sitemap. Just HTML.

File: budgetportal/urls.py [hunk 6]
```diff
@@ -42,19 +42,19 @@ def trigger_error(request):
...
         '/previews'
         '/(?P<sphere_slug>[\w-]+)'
         '/(?P<government_slug>[\w-]+)'
-        '/(?P<phase_slug>[\w-]+).json', cache_page(CACHE_SECS)(views.department_preview_json)),
+        '/(?P<phase_slug>[\w-]+).json', cache_page(CACHE_SECS)(views.department_preview_json), name='department-preview-json'),
 
     # Consolidated
     url(r'^json/(?P<financial_year_id>\d{4}-\d{2})'
-        '/consolidated.json', cache_page(CACHE_SECS)(views.consolidated_treemap_json)),
+        '/consolidated.json', cache_page(CACHE_SECS)(views.consolidated_treemap_json), name='consolidated'),
```
Comment (jbothma) [inline budgetportal/urls.py]:
```suggestion
        '/consolidated.json', cache_page(CACHE_SECS)(views.consolidated_treemap_json), name='consolidated-json'),
```

File: budgetportal/urls.py [hunk 7]
```diff
@@ -97,7 +97,7 @@ def trigger_error(request):
     url(r'^glossary/?$', views.glossary, name="glossary"),
     url(r'^faq/?$', views.faq, name="faq"),
     url(r'^guides/?$', views.guides, name="guides", kwargs={'slug': 'index'}),
-    url(r'^guides/(?P<slug>[-\w]+)/?$', views.guides, name="guides"),
+    url(r'^guides/(?P<slug>[-\w]+)/?$', views.guides, name="guide"),
```
Comment (jbothma) [inline budgetportal/urls.py]:
```suggestion
    url(r'^guides/(?P<slug>[-\w]+)/?$', views.guides, name="guide-list"),
```

File: budgetportal/sitemaps.py (hunk 5 above)
Comment (BarisSari) [reply budgetportal/sitemaps.py]: removed
https://github.com/vulekamali/datamanager/pull/185/commits/a09998b7e7d312745667d4696eeaa0878a379133

File: budgetportal/sitemaps.py (hunk 1 above)
Comment (BarisSari) [reply budgetportal/sitemaps.py]: removed

https://github.com/vulekamali/datamanager/pull/185/commits/a09998b7e7d312745667d4696eeaa0878a379133

File: budgetportal/sitemaps.py (hunk 2 above)
Comment (BarisSari) [reply budgetportal/sitemaps.py]: renamed.

https://github.com/vulekamali/datamanager/pull/185/commits/712af6608bdfc564221cbb2ce5545186a7b3e30a

**Pull request merged**